### PR TITLE
avoid trimming whitespace from cursor line on autosave

### PR DESCRIPTION
**Pull request by kevinushey**
### Intent

Addresses https://github.com/rstudio/rstudio/issues/14829.

### Approach

- Provide `isAutosaving()` to detect when a save is happening in response to our autosave timer
- During an auto-save, avoid trimming whitespace from lines containing the cursor / selection.

### Automated Tests

N/A

### QA Notes

Test via notes in https://github.com/rstudio/rstudio/issues/14829.

### Documentation

N/A

### Checklist

- [x] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [x] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [x] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [x] This PR passes all local unit tests
- [ ]

### CODE REVIEW

File: src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java [hunk 1]
```diff
@@ -3417,13 +3417,39 @@ private void fixupCodeBeforeSaving(Command ready)
...
-            Pattern pattern = Pattern.create("[ \t]+$");
-            String strippedCode = pattern.replaceAll(code, "");
+            String strippedCode = "";
+            
+            // If this is being performed as part of an autosave, avoid
+            // mutating lines containing the cursor.
+            if (isAutoSaving())
+            {
+               int startRow = docDisplay_.getSelectionStart().getRow();
+               int endRow = docDisplay_.getSelectionEnd().getRow();
```
Comment (gtritchie) [inline src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java:3435]: Does this detect the multi-line cursor mode (Ctrl+Alt+Up/Down Arrow) along with multi-line selection (I have no idea if that's the same thing to `getSelectionStart/End`)?

**Review by gtritchie — APPROVED** [review]
One question about multi-line cursors, but LGTM!

File: src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java (hunk 1 above)
Comment (kevinushey) [reply src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java:3435]: Good question -- it looks like this only applies to the "primary" cursor / selection. Should I do the legwork to support multiple cursors here?

Comment (gtritchie) [reply src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/TextEditingTarget.java:3435]: > Should I do the legwork to support multiple cursors here?

Probably not a very high priority, was just curious.

**Pull request merged**